ci-operator/step-registry/aws/deprovision/iam-profile/aws-deprovision-iam-profile-commands.sh

```diff
@@ -10,115 +10,79 @@ trap 'delete_all' EXIT TERM INT
+  aws_delete_role $REGION "$(head -n 1 ${SHARED_DIR}/aws_byo_role_name_worker)"
 
-	echo "Deleting policy ... "
-	aws --region $REGION iam delete-policy --policy-arn "$(head -n 1 ${SHARED_DIR}/aws_byo_policy_arn_master)"
-	aws --region $REGION iam delete-policy --policy-arn "$(head -n 1 ${SHARED_DIR}/aws_byo_policy_arn_worker)"
+  aws_delete_policy $REGION "$(head -n 1 ${SHARED_DIR}/aws_byo_policy_arn_master)"
+  aws_delete_policy $REGION "$(head -n 1 ${SHARED_DIR}/aws_byo_policy_arn_worker)"
 }
 
-master_profile=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:Name,Values=${INFRA_ID}-master*" | jq -r '.Reservations[].Instances[].IamInstanceProfile.Arn' | sort | uniq | awk -F '/' '{print $2}')
-worker_profile=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:Name,Values=${INFRA_ID}-worker*" | jq -r '.Reservations[].Instances[].IamInstanceProfile.Arn' | sort | uniq | awk -F '/' '{print $2}')
-
-master_role=$(aws --region $REGION iam get-instance-profile --instance-profile-name ${master_profile} | jq -r '.InstanceProfile.Roles[0].Arn' | awk -F '/' '{print $2}')
-worker_role=$(aws --region $REGION iam get-instance-profile --instance-profile-name ${worker_profile} | jq -r '.InstanceProfile.Roles[0].Arn' | awk -F '/' '{print $2}')
-
-master_policy_arn=$(aws --region $REGION iam list-attached-role-policies --role-name ${master_role} | jq -j '.AttachedPolicies[].PolicyArn')
-worker_policy_arn=$(aws --region $REGION iam list-attached-role-policies --role-name ${worker_role} | jq -j '.AttachedPolicies[].PolicyArn')
-
-ic_platform_profile=$(yq-go r "${CONFIG}" 'platform.aws.defaultMachinePlatform.iamProfile')
-ic_control_plane_profile=$(yq-go r "${CONFIG}" 'controlPlane.platform.aws.iamProfile')
-ic_compute_profile=$(yq-go r "${CONFIG}" 'compute[0].platform.aws.iamProfile')
-
-if ! is_empty "$ic_platform_profile"; then
-	aws --region $REGION iam get-instance-profile --instance-profile-name ${master_profile}
-	aws --region $REGION iam get-instance-profile --instance-profile-name ${worker_profile}
-	aws --region $REGION iam get-role --role-name ${master_role}
-	aws --region $REGION iam get-role --role-name ${worker_role}
-	aws --region $REGION iam get-policy --policy-arn ${master_policy_arn}
-	aws --region $REGION iam get-policy --policy-arn ${worker_policy_arn}
-fi
-
-if ! is_empty "$ic_control_plane_profile"; then
-	aws --region $REGION iam get-instance-profile --instance-profile-name ${master_profile}
-	aws --region $REGION iam get-role --role-name ${master_role}
-	aws --region $REGION iam get-policy --policy-arn ${master_policy_arn}
-fi
-
-if ! is_empty "$ic_compute_profile"; then
-	aws --region $REGION iam get-instance-profile --instance-profile-name ${worker_profile}
-	aws --region $REGION iam get-role --role-name ${worker_role}
-	aws --region $REGION iam get-policy --policy-arn ${worker_policy_arn}
-fi
+echo "Deleting BYO-IAM resources ..."
```

Comment:
So we will can not use this step to test if the BYO-iam-profile exists or not after clusters get destroyed.

Reply:
That's what I realized after asking you for approval. 
Fixed now, read IAM name from SHARED_DIR directly.
Thanks